Make github actions for amd64 and arm64

montgomery-auber opened the pull request:
I have successfully built images with Github actions and pushed to my account in docker hub. https://hub.docker.com/u/montgomeryauber   . I was not sure about removing the push to my account. Just tell me which account to replace it with, though I wont have the credentials.

williamdes commented on .github/workflows/build.yml:
```diff
@@ -1,7 +1,7 @@
-
 on: [push, pull_request]
 
+# https://github.com/roundcube/roundcubemail-docker/pull/200
```
```suggestion
```

williamdes commented on .github/workflows/build.yml:
```diff
@@ -25,11 +25,25 @@ jobs:
+          context: ${{ matrix.variant }}
+          platforms: linux/amd64,linux/arm64
+          push: true 
+          tags: montgomeryauber/${{ matrix.docker-tag }}:1.6,montgomeryauber/${{ matrix.docker-tag }}:latest
```
I think you can at least replace `montgomeryauber` by another matrix variable

williamdes commented on .github/workflows/build.yml:
```diff
@@ -25,11 +25,25 @@ jobs:
+        uses: docker/build-push-action@v3
+        with:
+          context: ${{ matrix.variant }}
+          platforms: linux/amd64,linux/arm64
```
Given that the list of archs of the base image is wide you should support as much archs too
See: https://hub.docker.com/_/php/tags?page=1&name=8.0-fpm

And here is the full list: https://github.com/sudo-bot/docker-phpmyadmin-snapshots/blob/28d025bff39dd1ded8d581a9e5c079fa3cb9fdf7/.github/workflows/publish.yml#L37

```suggestion
          platforms: linux/386,linux/amd64,linux/arm/v5,linux/arm/v7,linux/arm64/v8,linux/mips64le,linux/ppc64le,linux/s390x
```

williamdes commented on .github/workflows/build.yml:
```diff
@@ -37,13 +37,13 @@ jobs:
         with:
           context: ${{ matrix.variant }}
-          platforms: linux/amd64,linux/arm64
+          PLATFORM: "linux/386,linux/amd64,linux/arm/v5,linux/arm/v7,linux/arm64/v8,linux/mips64le,linux/ppc64le,linux/s390x,linux/arm64"
```
```suggestion
          platforms: "linux/386,linux/amd64,linux/arm/v5,linux/arm/v7,linux/arm64/v8,linux/mips64le,linux/ppc64le,linux/s390x,linux/arm64"
```

williamdes commented on .github/workflows/build.yml:
```diff
@@ -25,11 +25,25 @@ jobs:
+        uses: docker/build-push-action@v3
+        with:
+          context: ${{ matrix.variant }}
+          platforms: "linux/386,linux/amd64,linux/arm/v5,linux/arm/v7,linux/arm64/v8,linux/mips64le,linux/ppc64le,linux/s390x,linux/arm64"
```
I think some need to be removed because the composer images does not have them https://hub.docker.com/_/composer/tags

williamdes reviewed (approved):
Looks really nice!

montgomery-auber commented:
I had reduced it to just 2, though I am sure that all of the 386 and other traditional platforms are included. Does the Roundcube project have runners that support them all ? I’m just using the free runners.

 

 

> [linux/mips64le internal] load metadata for docker.io/library/composer:2:

------

Dockerfile:66

--------------------

  64 |                     rm -rf /var/lib/apt/lists/*

  65 |     

  66 | >>> COPY --from=composer:2 /usr/bin/composer /usr/bin/composer

  67 |     

  68 |     RUN a2enmod rewrite

--------------------

ERROR: failed to solve: composer:2: no match for platform in manifest sha256:6fb16d897d4376d711880522b13743c8375b29d4c57183555b58c5be751fc164: not found

Error: buildx failed with: ERROR: failed to solve: composer:2: no match for platform in manifest sha256:6fb16d897d4376d711880522b13743c8375b29d4c57183555b58c5be751fc164: not found

 

 

 

 

 

From: William Desportes ***@***.***> 
Sent: Sunday, December 11, 2022 8:22 PM
To: roundcube/roundcubemail-docker ***@***.***>
Cc: Montgomery Auber ***@***.***>; Author ***@***.***>
Subject: Re: [roundcube/roundcubemail-docker] Make github actions for amd64 and arm64 (PR #201)

 

@williamdes commented on this pull request.

  _____  

In .github/workflows/build.yml <https://github.com/roundcube/roundcubemail-docker/pull/201#discussion_r1045276884> :

> @@ -25,11 +25,25 @@ jobs:
         with:
           username: ${{ secrets.DOCKER_PULL_USERNAME }}
           password: ${{ secrets.DOCKER_PULL_PASSWORD }}
-      - name: Build image variant "${{ matrix.variant }}"
-        run: cd ${{ matrix.variant }} && docker buildx build ./ -t ${{ matrix.docker-tag }}
+
+      - name: Set up QEMU
+        uses: ***@***.***
+      - name: Set up Docker Buildx
+        uses: ***@***.***
+        with:
+          buildkitd-flags: --debug
+
+      - name: Build and push image for "${{ matrix.variant }}"
+        uses: ***@***.***
+        with:
+          context: ${{ matrix.variant }}
+          platforms: "linux/386,linux/amd64,linux/arm/v5,linux/arm/v7,linux/arm64/v8,linux/mips64le,linux/ppc64le,linux/s390x,linux/arm64"

I think some need to be removed because the composer images does not have them https://hub.docker.com/_/composer/tags

—
Reply to this email directly, view it on GitHub <https://github.com/roundcube/roundcubemail-docker/pull/201#pullrequestreview-1212737430> , or unsubscribe <https://github.com/notifications/unsubscribe-auth/AL4QT72GCCA5UUHP44ZBIMLWMYLTVANCNFSM6AAAAAASXDOVUM> .
You are receiving this because you authored the thread.  <https://github.com/notifications/beacon/AL4QT7456ALZEAYT4ZTMOTTWMYLTVA5CNFSM6AAAAAASXDOVUOWGG33NNVSW45C7OR4XAZNRKB2WY3CSMVYXKZLTORJGK5TJMV32UY3PNVWWK3TUL5UWJTSIJDTZM.gif> Message ID: ***@***.*** ***@***.***> >

montgomery-auber commented:
Thank you

 

From: William Desportes ***@***.***> 
Sent: Sunday, December 11, 2022 8:22 PM
To: roundcube/roundcubemail-docker ***@***.***>
Cc: Montgomery Auber ***@***.***>; Author ***@***.***>
Subject: Re: [roundcube/roundcubemail-docker] Make github actions for amd64 and arm64 (PR #201)

 

@williamdes approved this pull request.

Looks really nice!

—
Reply to this email directly, view it on GitHub <https://github.com/roundcube/roundcubemail-docker/pull/201#pullrequestreview-1212737460> , or unsubscribe <https://github.com/notifications/unsubscribe-auth/AL4QT73IQU6GPG4RSJSC2SLWMYLU5ANCNFSM6AAAAAASXDOVUM> .
You are receiving this because you authored the thread.  <https://github.com/notifications/beacon/AL4QT77LXG3CFOQ4FAHXFFDWMYLU5A5CNFSM6AAAAAASXDOVUOWGG33NNVSW45C7OR4XAZNRKB2WY3CSMVYXKZLTORJGK5TJMV32UY3PNVWWK3TUL5UWJTSIJDT3I.gif> Message ID: ***@***.***>

williamdes commented:
> I had reduced it to just 2, though I am sure that all of the 386 and other traditional platforms are included. Does the Roundcube project have runners that support them all ? I’m just using the free runners.


It has nothing to do with roundcube or runners :) 
It is purely based on tags of images you use 
In our case the php image and the composer image 
So you need to reduce to the archs supported by both images

By the way the runners are hosted and run by GitHub on our case 

Docker buildx can build any arch on the processor of the machine (most probably x64)

thomascube commented:
Hi @montgomery-auber and @williamdes
Thanks a lot for your efforts! As I'm not really familiar with Github actions yet, I can't add much to the conversation. The us of the `DOCKER_PULL_USERNAME` secret for the Docker hub repository name is OK but we could also just hard-code it to "roundcube". I have also create a new secret named `DOCKER_PUSH_PASSWORD` with an access token to push images. The `DOCKER_PULL_PASSWORD` has been changed to a token with read-only privileges.

thomascube commented:
For actually publishing images to Docker hub, we need to adapt the image names and tags to match the current naming schema. Also, image pushes should only happen on "push" events (or maybe tagging) but not for pull requests. For the latter, only tests should be executed.

montgomery-auber commented:
Sorry for the delay, I hadn't realized at the time that I need to do more stuff. I am planning to pull official images instead of my test images for arm for [https://aws.amazon.com/marketplace/pp/prodview-7lthtvl7jhclu](https://aws.amazon.com/marketplace/pp/prodview-7lthtvl7jhclu)
Plus the ARM version works on my raspberry pi !
Can't wait to help out more.

williamdes commented on .github/workflows/build.yml:
```diff
@@ -25,8 +25,21 @@ jobs:
+          context: ${{ matrix.variant }}
+          platforms: "linux/386,linux/amd64,linux/arm/v7,linux/arm64/v8,linux/ppc64le,linux/s390x,linux/arm64"
+          push: true 
+          tags: ${{ matrix.docker-tag }}
```
I think it would fail because it's missing the image name

montgomery-auber commented:
I am trying to figure out how to tag and have the image pushed to docker hub. How was it previously pushed to docker hub.  ${{ matrix.docker-tag }} in [build.yaml](https://github.com/roundcube/roundcubemail-docker/blob/master/.github/workflows/build.yml)    refers to names like roundcube-test-fpm. But the image name and tags that I see in docker hub [tags](https://hub.docker.com/r/roundcube/roundcubemail/tags)  seem to come from somewhere else. Were they manually uploaded ? 

I will try to make my push look like the image names and tags in docker hub. I hope to get it right

williamdes commented:
> I am trying to figure out how to tag and have the image pushed to docker hub. How was it previously pushed to docker hub. ${{ matrix.docker-tag }} in [build.yaml](https://github.com/roundcube/roundcubemail-docker/blob/master/.github/workflows/build.yml) refers to names like roundcube-test-fpm. But the image name and tags that I see in docker hub [tags](https://hub.docker.com/r/roundcube/roundcubemail/tags) seem to come from somewhere else. Were they manually uploaded ?
> 
> I will try to make my push look like the image names and tags in docker hub. I hope to get it right

Yeah I think they are manually uploaded

The best would be to add the registry name to the tag in the build matrix

williamdes commented on .github/workflows/build.yml:
```diff
@@ -13,9 +12,10 @@ jobs:
           - { variant: 'apache', test-files: ['apache-postgres'], docker-tag: 'roundcube-test-apache' }
           - { variant: 'fpm', test-files: ['fpm-postgres'], docker-tag: 'roundcube-test-fpm' }
           - { variant: 'fpm-alpine', test-files: ['fpm-postgres'], docker-tag: 'roundcube-test-fpm-alpine' }
```
```suggestion
          - { variant: 'apache', test-files: ['apache-postgres'], docker-tag: 'roundcube/roundcubemail:roundcube-test-apache' }
          - { variant: 'fpm', test-files: ['fpm-postgres'], docker-tag: 'roundcube/roundcubemail:roundcube-test-fpm' }
          - { variant: 'fpm-alpine', test-files: ['fpm-postgres'], docker-tag: 'roundcube/roundcubemail:roundcube-test-fpm-alpine' }
```

montgomery-auber commented:
HI there, I am having issues with the multi arch. Have succeeded only with amd32 and 64 and arm 32 and 64, the others are giving issues like this with "provenance".
ERROR: failed to solve: missing provenance for ps2js6jymdyr5v6zhe6p1vd5s
Error: buildx failed with: ERROR: failed to solve: missing provenance for ps2js6jymdyr5v6zhe6p1vd5s
Please follow my other branch that isn't part of the PR, as it is pushing to my docker hub. Once all is fine I will manually change the username (instead of using a var) to roundcube. I have successfully pushed all the tags to one repo, 
My branch https://github.com/montgomery-auber/roundcubemail-docker/tree/correct-tags   "correct-tags" 
See my docker hub account. https://hub.docker.com/repository/docker/montgomeryauber/roundcubemail/tags?page=1&ordering=last_updated

williamdes commented:
> ERROR: failed to solve: missing provenance for ps2js6jymdyr5v6zhe6p1vd5s

Can you post a mini to a build or some context at where was the build before it failed?

Apart from that if it builds and pushes it's perfect 🎉

montgomery-auber commented:
https://github.com/montgomery-auber/roundcubemail-docker/actions/runs/3931723114/jobs/6723348916
ERROR: failed to solve: composer:2: no match for platform in manifest sha256:34da7f25f8f1582b34861afd6c6e780b4f5cc571ffa0e94b91754d403a2d4371: not found
Error: buildx failed with: ERROR: failed to solve: composer:2: no match for platform in manifest sha256:34da7f25f8f1582b34861afd6c6e780b4f5cc571ffa0e94b91754d403a2d4371: not found

I added first linux/mips64le , so it would die in the beginning. I will try others. Seems like Github might limit platforms that docker desktop does support

montgomery-auber commented:
These 2 platforms did not work, linux/arm/v5 AND linux/mips64le  all the others work. I will change the username tomorrow

williamdes commented on .github/workflows/build.yml:
```diff
@@ -25,15 +30,31 @@ jobs:
+        uses: docker/build-push-action@v3
+        with:
+          context: ${{ matrix.variant }}
+          platforms: "linux/s390x,linux/ppc64le,linux/arm,linux/386,linux/amd64,linux/arm64/v8"
```
you can move the comment here

williamdes commented on tests/docker-compose.test-apache-postgres.yml:
```diff
@@ -2,7 +2,8 @@ version: "2"
 
 services:
   roundcubemail:
-    image: ${ROUNDCUBEMAIL_TEST_IMAGE:-roundcube-test-apache}
```
is adding this `ROUNDCUBEMAIL_TEST_IMAGE` to GitHub secrets not working ?

williamdes commented:
> These 2 platforms did not work, linux/arm/v5 AND linux/mips64le all the others work. I will change the username tomorrow

That's okay. Most people will be happy, it's great work

Maybe you can squash all your commits and force push it. Please do not close this PR ;)

williamdes commented:
> https://github.com/montgomery-auber/roundcubemail-docker/actions/runs/3931723114/jobs/6723348916 ERROR: failed to solve: composer:2: no match for platform in manifest sha256:34da7f25f8f1582b34861afd6c6e780b4f5cc571ffa0e94b91754d403a2d4371: not found Error: buildx failed with: ERROR: failed to solve: composer:2: no match for platform in manifest sha256:34da7f25f8f1582b34861afd6c6e780b4f5cc571ffa0e94b91754d403a2d4371: not found
> 
> I added first linux/mips64le , so it would die in the beginning. I will try others. Seems like Github might limit platforms that docker desktop does support

Nope, as I said earlier it is limited to the one that supports the least platforms. And it is composer that is not having mips64le or armv5 

See: https://hub.docker.com/_/composer/tags

You can put that in your comment on the workflow :)

montgomery-auber commented:
Ok, Now I understand, composer is in charge of the ROundcube plugins, like password change and lots more stuff, and they don't support the 2 platforms. I now added linux/arm/v7, and v6 , hopefully it won't fail.

See: https://hub.docker.com/_/composer/tags

You can put that in your comment on the workflow :)

> > https://github.com/montgomery-auber/roundcubemail-docker/actions/runs/3931723114/jobs/6723348916 ERROR: failed to solve: composer:2: no match for platform in manifest sha256:34da7f25f8f1582b34861afd6c6e780b4f5cc571ffa0e94b91754d403a2d4371: not found Error: buildx failed with: ERROR: failed to solve: composer:2: no match for platform in manifest sha256:34da7f25f8f1582b34861afd6c6e780b4f5cc571ffa0e94b91754d403a2d4371: not found
> > I added first linux/mips64le , so it would die in the beginning. I will try others. Seems like Github might limit platforms that docker desktop does support
> 
> Nope, as I said earlier ([#201 (comment)](https://github.com/roundcube/roundcubemail-docker/pull/201#issuecomment-1345629795)) it is limited to the one that supports the least platforms. And it is composer that is not having mips64le or armv5
> 
> See: https://hub.docker.com/_/composer/tags
> 
> You can put that in your comment on the workflow :)

williamdes commented on .github/workflows/build.yml:
```diff
@@ -25,8 +30,22 @@ jobs:
+          platforms: "linux/arm/v6,linux/arm/v7,linux/s390x,linux/ppc64le,linux/386,linux/amd64,linux/arm64/v8"
+          push: true 
+          tags: ${{ matrix.docker-tag }}
+          # does not work linux/arm/v5 AND linux/mips64le - composer does not support  mips64le or armv5 although docker buildx does support them
```
```suggestion
          # does not work linux/arm/v5 AND linux/mips64le - composer does not support  mips64le or armv5 nor does the php image support them on the alpine variant
```

So the two are supported only on PHP non Alpine images, see: https://hub.docker.com/_/php/tags?page=1&name=8.0-fpm
See: https://hub.docker.com/_/php/tags?page=1&name=8.0-fpm-alpine

williamdes replied:
For reference:

PHP:
![image](https://user-images.githubusercontent.com/7784660/212985828-2f28ae8e-1e2a-4411-b3be-6bb9de81ae7b.png)
![image](https://user-images.githubusercontent.com/7784660/212985854-536ed956-2b5b-4708-baae-68d90a457179.png)

Composer:
![image](https://user-images.githubusercontent.com/7784660/212986013-9cf07e1b-39ec-4c09-94e9-431a82ce8549.png)

williamdes commented on .github/workflows/build.yml:
```diff
@@ -25,8 +30,22 @@ jobs:
+        uses: docker/build-push-action@v3
+        with:
+          context: ${{ matrix.variant }}
+          platforms: "linux/arm/v6,linux/arm/v7,linux/s390x,linux/ppc64le,linux/386,linux/amd64,linux/arm64/v8"
```
```suggestion
          platforms: "linux/arm/v6,linux/arm/v7,linux/s390x,linux/ppc64le,linux/386,linux/amd64"
```

`linux/arm64/v8` is not supported by composer or alpine PHP images

montgomery-auber replied:
removing the v8 now

williamdes replied:
It's possible I am mistaken on this because they might be the same between `linux/arm64/v8` and `linux/arm64`.
Did builds fail on it ?

montgomery-auber commented:
Finally ready to change the username to roundcube/roundcubemail:TAGS , However, I see that there is also an account roundcubeorg . So now they can be pushed automatically to the roundcube account at dockerhub. Be sure to fill in the following in github actions secrets.
{{ secrets.DOCKER_PULL_USERNAME }}
{{ secrets.DOCKER_PULL_PASSWORD }}

williamdes commented on tests/docker-compose.test-fpm-postgres.yml:
```diff
@@ -2,7 +2,9 @@ version: "2"
 services:
   roundcubemail-fpm:
-    image: ${ROUNDCUBEMAIL_TEST_IMAGE:-roundcube-test-fpm}
+    #image: montgomeryauber/roundcubemail:1.6.0-fpm
+    #image: ${ROUNDCUBEMAIL_TEST_IMAGE:-roundcube-test-fpm}
+    image: ${ROUNDCUBEMAIL_TEST_IMAGE}
```
can you clean this left over and squash all commits ?

williamdes reviewed (approved):
Thank you for your great work !

montgomery-auber commented:
I updated the test  pull image to hardcoded

thomascube commented:
Thanks a lot for all your work! If I read this correctly (`push: true`), running this action will push the build images to docker hub. This is probably not what we want to do `on: [push, pull_request]`. Could we split this into different actions to
1. build and test on pull_request and push (no need for multi-arch)
2. build and push either on tag or workflow_dispatch (manual action)

williamdes commented:
> Thanks a lot for all your work! If I read this correctly (`push: true`), running this action will push the build images to docker hub. This is probably not what we want to do `on: [push, pull_request]`. Could we split this into different actions to
> 
>     1. build and test on pull_request and push (no need for multi-arch)
> 
>     2. build and push either on tag or workflow_dispatch (manual action)

I just want to add a technical aspect, you do not have the choice with multi arch images. You must push them because the local registry can not handle them as the manifest is multi arch. There is probably a way to make it build without a push but keep in mind what I said above. That's why you need to specify only let's say amd64 like your workstation and then you can change push to `--load` understand that it loads the built image into the local registry

This is a bare bones cli usage that illustrates the two modes
https://github.com/sudo-bot/nut-upsd/blob/013caefc1c2cb1f6de0f3e06839d111cd65d15e1/Makefile#L18


To conclude I think point 1 can do multi arch specifying only amd64 and load action

And point 2 use the work that has been done here

@thomascube can you squash merge this work so I can do the final points of it?

montgomery-auber commented:
I think that we should use this as/is or make minor changes. INstead of making the push manual, as @williamdes said.


> > Thanks a lot for all your work! If I read this correctly (`push: true`), running this action will push the build images to docker hub. This is probably not what we want to do `on: [push, pull_request]`. Could we split this into different actions to
> > ```
> > 1. build and test on pull_request and push (no need for multi-arch)
> > 
> > 2. build and push either on tag or workflow_dispatch (manual action)
> > ```
> 
> I just want to add a technical aspect, you do not have the choice with multi arch images. You must push them because the local registry can not handle them as the manifest is multi arch. There is probably a way to make it build without a push but keep in mind what I said above. That's why you need to specify only let's say amd64 like your workstation and then you can change push to `--load` understand that it loads the built image into the local registry
> 
> This is a bare bones cli usage that illustrates the two modes https://github.com/sudo-bot/nut-upsd/blob/013caefc1c2cb1f6de0f3e06839d111cd65d15e1/Makefile#L18
> 
> To conclude I think point 1 can do multi arch specifying only amd64 and load action
> 
> And point 2 use the work that has been done here
> 
> @thomascube can you squash merge this work so I can do the final points of it?

williamdes commented:
Hi @thomascube 
Can we get this merged to be tried on [1.6.1](https://github.com/roundcube/roundcubemail/releases/tag/1.6.1) that is not yet on docker hub ?

thomascube commented:
@williamdes I just approved the actions and let it run on the 1.6.0 version which needs an update of the base image anyways. Will merge it afterwards...

thomascube commented:
@williamdes The runs failed with missing authentication to Docker Hub although the according secrets are set... maybe you see what's missing.

williamdes commented:
> @williamdes The runs failed with missing authentication to Docker Hub although the according secrets are set... maybe you see what's missing.

Yes, because we use `docker/build-push-action` and not the same action that I use on this workflow: https://github.com/sudo-bot/docker-phpmyadmin-snapshots/blob/main/.github/workflows/publish.yml

It's probably using a container action and not a other type of action. If you can do some copy pasting of my workflow it should work out of the box. Else let me know here and I will do the changes for you :)

thomascube commented:
When running from master branch, the login suceeds, however the action used the wrong secret and was not allowed to push. Also there are a few deprecation warnings we should address. Now running again: https://github.com/roundcube/roundcubemail-docker/actions/runs/4058059746

thomascube commented:
@williamdes, @montgomery-auber  The actions have now run successfully from the master branch and images on Docker hub have been updated: https://hub.docker.com/repository/docker/roundcube/roundcubemail/tags?page=1&ordering=last_updated&name=latest
Nice work!

williamdes commented:
Awesome!!

montgomery-auber commented:
Thanks! I am really happy to have contributed. I hope to contribute more in the future. This is part of complete dockerized email  project including certbot certs. I was using the image in my own docker hub account, but will now switch to pulling the official. 

I still use the image as base, since the logs don't go to stdout and stderr for docker logs, I opened a case. But otherwise all is perfect. 

**Yay! for the universe that Roundcube docker** can run on so many more devices with a simple pull. Raspbery Pi and all replicas, like La potatoe and Orange Pi launch the container and send / receive IMAP / SMTP securely. It also runs perfectly well on AWS EC2 graviton instances, I have been testing a lot with these.

thomascube commented:
Hello @montgomery-auber and @williamdes
One last question to you guys: I'd like to announce the multi-arch feature on roundcube.net and of course I'd like to give the according credits to you. Is it OK for you if I mention your names with links to your Github profile there?

williamdes commented:
> Hello @montgomery-auber and @williamdes One last question to you guys: I'd like to announce the multi-arch feature on roundcube.net and of course I'd like to give the according credits to you. Is it OK for you if I mention your names with links to your Github profile there?

yeah, sure, my github profile link is perfect 
can we have the message here before posting so we maybe add some details?

montgomery-auber commented:
Hi, That would be great, My new linkedin and / or github is great!  https://www.linkedin.com/in/montgomery-auber-658405259/

thomascube commented:
Thanks so far! Here's the draft: https://github.com/roundcube/roundcube.github.com/blob/preview/multi-arch-docker-article/_posts/2023-02-07-multi-arch-docker-images.md

williamdes commented:
> Thanks so far! Here's the draft: https://github.com/roundcube/roundcube.github.com/blob/preview/multi-arch-docker-article/_posts/2023-02-07-multi-arch-docker-images.md

Awesome, I checked the arch list and it's okay.
I'd add a sentence that it's limited to the archs available on https://hub.docker.com/_/php and https://hub.docker.com/_/composer

Thank you for this nice post !

williamdes commented:
Hi @thomascube 
What is `roundcubeorg/roundcubemail` on Docker Hub ?
It's not multi arch, is it up to date ?

thomascube commented:
@williamdes `roundcubeorg` was created as an organizational unit in order to get (some) free services from Dockerhub for open source projects. We used it to auto-build images on changes to this repo. I usually copied the built images over to `roundcubemail` because this is where we used to publish our images. Now that we build on Github, we'll probably close `roundcubeorg` again.

williamdes commented:
@thomascube you can do a bit simpler that this:

> Now that we build on Github, we'll probably close roundcubeorg again.

Just disable auto builds, and add a push command when we push the built image to also push it to the other org?
